src/main/java/org/terasology/metalrenegades/world/dynamic/discoverables/DiscoverablesRasterizer.java

```diff
@@ -30,19 +35,37 @@
      */
     private Block chest;
 
+    private Prefab structure;
+
     @Override
     public void initialize() {
         chest = CoreRegistry.get(BlockManager.class).getBlock("MetalRenegades:hiddenChest.RIGHT");
+        structure = Objects.requireNonNull(CoreRegistry.get(PrefabManager.class)).getPrefab("MetalRenegades:wellTemplate");
+
         entityManager = CoreRegistry.get(EntityManager.class);
     }
 
     @Override
     public void generateChunk(CoreChunk chunk, Region chunkRegion) {
         DiscoverablesFacet discoverablesFacet = chunkRegion.getFacet(DiscoverablesFacet.class);
 
+        SpawnBlockRegionsComponent spawnBlockRegionsComponent =
+                structure.getComponent(SpawnBlockRegionsComponent.class);
+
         for (Map.Entry<BaseVector3i, DiscoverablesChest> entry : discoverablesFacet.getWorldEntries().entrySet()) {
-            Vector3i chestPosition = new Vector3i(entry.getKey());
-            chunk.setBlock(ChunkMath.calcRelativeBlockPos(chestPosition), chest);
+            Vector3i structurePosition = new Vector3i(entry.getKey());
+
+            for (SpawnBlockRegionsComponent.RegionToFill regionToFill : spawnBlockRegionsComponent.regionsToFill) {
+                  Block block = regionToFill.blockType;
+
+                  Region3i region = regionToFill.region;
```

Comment:
`Region3i` is deprecated. Use `BlockRegion` instead

Reply:
This part was accidentally included as part of the structure template updates, sorry about that!